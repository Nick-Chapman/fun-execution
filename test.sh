#/bin/bash

export PATH=exe

run() {
    echo '--------------------------------------------------'
    echo "$@"
    echo '--------------------------------------------------'
    $@
}
run thrice-thrice.exe
run nfib.exe
run triangle.exe
run fact.exe 6
run nthPrime.exe
run combinator-fact.exe
run list-processing.exe
run over.exe
run pap-over-app.exe

# Need to print string values properly to enable this...
#run pythagorian.exe

exit 0
