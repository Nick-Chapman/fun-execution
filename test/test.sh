#/bin/bash

export PATH=_build/exe

run() {
    echo '--------------------------------------------------'
    echo "$@"
    echo '--------------------------------------------------'
    $@
}

run fact 6
run triangle 100
run nfib 10
run nthPrime 10

run thrice-thrice
run combinator-fact
run list-processing
run over
run pap-over-app

# Need to print string values properly to enable this...
#run pythagorian.exe

exit 0
