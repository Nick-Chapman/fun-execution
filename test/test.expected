--------------------------------------------------
fact 6
--------------------------------------------------
the final result is: 720
heap used, 20 cells
#steps = 196
--------------------------------------------------
triangle 100
--------------------------------------------------
the final result is: 5050
heap used, 7 cells
#steps = 2244
--------------------------------------------------
nfib 10
--------------------------------------------------
the final result is: 177
heap used, 618 cells
#steps = 5308
--------------------------------------------------
nthPrime 10
--------------------------------------------------
the final result is: 29
heap used, 2675 cells
#steps = 16671
--------------------------------------------------
thrice-thrice
--------------------------------------------------
the final result is: -27
heap used, 0 cells
#steps = 148
--------------------------------------------------
combinator-fact
--------------------------------------------------
the final result is: 720
heap used, 9545 cells
#steps = 62072
--------------------------------------------------
list-processing
--------------------------------------------------
the final result is: 338450
heap used, 1825 cells
#steps = 11515
--------------------------------------------------
over
--------------------------------------------------
the final result is: 200
heap used, 11 cells
#steps = 52
--------------------------------------------------
pap-over-app
--------------------------------------------------
stack exhausted